### PR TITLE
Consul connect

### PR DESCRIPTION
**Pull request by gabriel-v**
Everything almost works; there's one problem that keeps popping up.

When running the example job (`example/connect.nomad` taken from the [Connect Guide](https://www.nomadproject.io/guides/integrations/consul-connect/index.html#run-the-connect-enabled-services)) envoy fails to run with the following:

```[2019-11-11 15:21:23.664][1][info][main] [source/server/server.cc:238] initializing epoch 0 (hot restart version=11.104)
[2019-11-11 15:21:23.664][1][info][main] [source/server/server.cc:240] statically linked extensions:
[2019-11-11 15:21:23.664][1][info][main] [source/server/server.cc:242]   access_loggers: envoy.file_access_log,envoy.http_grpc_access_log
[2019-11-11 15:21:23.664][1][info][main] [source/server/server.cc:245]   filters.http: envoy.buffer,envoy.cors,envoy.csrf,envoy.ext_authz,envoy.fault,envoy.filters.http.dynamic_forward_proxy,envoy.filters.http.grpc_http1_reverse_bridge,envoy.filters.http.header_to_metadata,envoy.filters.http.jwt_authn,envoy.filters.http.original_src,envoy.filters.http.rbac,envoy.filters.http.tap,envoy.grpc_http1_bridge,envoy.grpc_json_transcoder,envoy.grpc_web,envoy.gzip,envoy.health_check,envoy.http_dynamo_filter,envoy.ip_tagging,envoy.lua,envoy.rate_limit,envoy.router,envoy.squash
[2019-11-11 15:21:23.664][1][info][main] [source/server/server.cc:248]   filters.listener: envoy.listener.original_dst,envoy.listener.original_src,envoy.listener.proxy_protocol,envoy.listener.tls_inspector
[2019-11-11 15:21:23.664][1][info][main] [source/server/server.cc:251]   filters.network: ```
envoy.client_ssl_auth,envoy.echo,envoy.ext_authz,envoy.filters.network.dubbo_proxy,envoy.filters.network.mysql_proxy,envoy.filters.network.rbac,envoy.filters.network.sni_cluster,envoy.filters.network.thrift_proxy,envoy.filters.network.zookeeper_proxy,envoy.http_connection_manager,envoy.mongo_proxy,envoy.ratelimit,envoy.redis_proxy,envoy.tcp_proxy
[2019-11-11 15:21:23.664][1][info][main] [source/server/server.cc:253]   stat_sinks: envoy.dog_statsd,envoy.metrics_service,envoy.stat_sinks.hystrix,envoy.statsd
[2019-11-11 15:21:23.664][1][info][main] [source/server/server.cc:255]   tracers: envoy.dynamic.ot,envoy.lightstep,envoy.tracers.datadog,envoy.tracers.opencensus,envoy.zipkin
[2019-11-11 15:21:23.664][1][info][main] [source/server/server.cc:258]   transport_sockets.downstream: envoy.transport_sockets.alts,envoy.transport_sockets.tap,raw_buffer,tls
[2019-11-11 15:21:23.664][1][info][main] [source/server/server.cc:261]   transport_sockets.upstream: envoy.transport_sockets.alts,envoy.transport_sockets.tap,raw_buffer,tls
[2019-11-11 15:21:23.664][1][info][main] [source/server/server.cc:267] buffer implementation: old (libevent)
[2019-11-11 15:21:23.664][1][debug][misc] [source/common/filesystem/posix/filesystem_impl.cc:93] Unable to determine canonical path for /secrets/envoy_bootstrap.json: No such file or directory
[2019-11-11 15:21:23.664][1][critical][main] [source/server/server.cc:93] error initializing configuration '/secrets/envoy_bootstrap.json': Invalid path: /secrets/envoy_bootstrap.json
[2019-11-11 15:21:23.664][1][info][main] [source/server/server.cc:560] exiting
[2019-11-11 15:21:23.664][1][debug][init] [source/common/init/watcher_impl.cc:27] init manager Server destroyed
Invalid path: /secrets/envoy_bootstrap.json
```

I checked the path on `./var/nomad/allocs/<ID>/`: inside the docker container they are owned by `nobody:nogroup`, but outside the container they are owned by `root:root`.

### CODE REVIEW

File: docs/Manual-Installation.md [hunk 1]
```diff
@@ -22,6 +22,17 @@ This guide assumes a recent Debian/Ubuntu installation with Python 3.6+ and `pip
...
 
+
+* Download CNI Plugins as [instructed by the Nomad docs](https://www.nomadproject.io/guides/integrations/consul-connect/index.html#cni-plugins):
+
+
+    ```bash
+    curl -L -o cni-plugins.tgz https://github.com/containernetworking/plugins/releases/download/v0.8.1/cni-plugins-linux-amd64-v0.8.1.tgz
+    sudo mkdir -p /opt/cni/bin
+    sudo tar -C /opt/cni/bin -xzf cni-plugins.tgz
+    rm -f cli-plugins.tgz
```
Comment (mgax) [inline docs/Manual-Installation.md]: typo, `cli` or `cni`?

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -21,4 +21,5 @@ chown -R $USERID:$GROUPID ./var
 
 python3 cluster.py configure-network
 
-exec sudo -nHu vagrant DOCKER_BIN=$DOCKER_BIN python3 ./cluster.py supervisord
+export USER=vagrant
+python3 ./cluster.py supervisord
```
Comment (mgax) [inline docker-entrypoint.sh]: nice

File: Dockerfile [hunk 3]
```diff
@@ -18,6 +18,14 @@ RUN set -e \
  && apt-get clean && rm -rf /var/lib/apt/lists/* \
  && mkdir -p /app/var && mkdir -p /app/bin
 
+RUN sed -i 's/^%sudo.*$/%sudo ALL=(ALL) NOPASSWD: ALL/' /etc/sudoers
```
Comment (mgax) [inline Dockerfile]: Or just create a file in `/etc/sudoers.d/`, easier to understand.

File: ci/test-host.sh [hunk 4]
```diff
@@ -1,5 +1,13 @@
 #!/bin/bash -ex
 
+echo "installing consul connect requirements"
+cd /tmp
+curl -L -o cni-plugins.tgz https://github.com/containernetworking/plugins/releases/download/v0.8.1/cni-plugins-linux-amd64-v0.8.1.tgz
+sudo mkdir -p /opt/cni/bin
+sudo tar -C /opt/cni/bin -xzf cni-plugins.tgz
+rm -f cli-plugins.tgz
```
Comment (mgax) [inline ci/test-host.sh]: here also, `cli` or `cni`?

File: Dockerfile (hunk 3 above)
Comment (gabriel-v) [reply Dockerfile]: And hunt down the already-existing %sudo line to replace it?  Not easier.

We could forego this entirely, I don't think we need it in this patch.

File: docker-entrypoint.sh (hunk 2 above)
Comment (gabriel-v) [reply docker-entrypoint.sh]: The `sudo` dissapearing means my signal hacks don't work anymore for `docker stop`, so I updated the shutdown sequence.

**Comment by gabriel-v** [comment]
maybe later

**Pull request closed without merging**